src/libraries/System.Linq/src/System/Linq/Min.cs

```diff
@@ -446,7 +443,7 @@ public static decimal Min(this IEnumerable<decimal> source)
                     while (e.MoveNext())
                     {
                         TSource x = e.Current;
-                        if (comparer.Compare(x, value) < 0)
+                        if (Comparer<TSource>.Default.Compare(x, value) < 0)
```

Comment:
Would it be possible to explain why you cannot use a local for the comparer in order to have this optimalization kick in?
Thx!

Reply:
The JIT doesn't track the concrete type of an instance stored in every local like it does for statics.